ArcGISRuntimeSDKQt_CppSamples/EditData/EditWithBranchVersioning/EditWithBranchVersioning.cpp

```diff
@@ -0,0 +1,377 @@
+    if (m_selectedFeature)
+    {
+      if (m_serviceGeodatabase->versionName() == m_serviceGeodatabase->defaultVersionName())
+      {
+        clearSelectedFeature();
+        m_busy = false;
+        emit busyChanged();
+        return;
+      }
+      const Point clickedPoint = m_mapView->screenToLocation(mouseEvent.x(), mouseEvent.y());
+      moveFeature(clickedPoint);
+    }
+    else
+    {
+      // call identify on the map view
+      m_mapView->identifyLayer(m_featureLayer, mouseEvent.x(), mouseEvent.y(), 5, false, 1);
+    }
+  });
+
+  connect(m_mapView, &MapQuickView::viewpointChanged, this, [this]()
+  {
+    if (!m_featureLayer)
+      return;
+
+    m_featureLayer->clearSelection();
+    clearSelectedFeature();
+    emit hideWindow();
+  });
+
+  // connect to the identifyLayerCompleted signal on the map view
+  connect(m_mapView, &MapQuickView::identifyLayerCompleted, this, [this](QUuid, IdentifyLayerResult* identifyResult)
+  {
+    if (!identifyResult)
+    {
+      clearSelectedFeature();
+      return;
+    }
+    if (!identifyResult->geoElements().empty())
+    {
+      m_selectedFeature = static_cast<ArcGISFeature*>(identifyResult->geoElements().at(0));
+      // select the item in the result
+      m_featureLayer->selectFeature(m_selectedFeature);
+
+      // Obtain placename attribute for the callout title
+      const QString featureName = m_selectedFeature->attributes()->attributeValue("PLACENAME").toString();
+      m_mapView->calloutData()->setTitle(QString("<br><font size=\"+2\">%1</font>").arg(featureName));
+      m_mapView->calloutData()->setLocation(m_selectedFeature->geometry().extent().center());
+
+      m_currentTypeDamage = m_selectedFeature->attributes()->attributeValue("TYPDAMAGE").toString();
+      emit currentTypeDamageChanged();
+      emit featureSelected();
+    }
+    m_busy = false;
+    emit busyChanged();
+  });
+}
+
+void EditWithBranchVersioning::connectSgdbSignals()
+{
+  m_serviceGeodatabase = new ServiceGeodatabase(QUrl("https://sampleserver7.arcgisonline.com/arcgis/rest/services/DamageAssessment/FeatureServer"), this);
+  m_busy = true;
+  emit busyChanged();
+
+  connect(m_serviceGeodatabase, &ServiceGeodatabase::doneLoading, this, &EditWithBranchVersioning::onSgdbDoneLoadingCompleted);
+
+  connect(m_serviceGeodatabase, &ServiceGeodatabase::createVersionCompleted, this, &EditWithBranchVersioning::onCreateVersionCompleted);
+
+  connect(m_serviceGeodatabase, &ServiceGeodatabase::applyEditsCompleted, this, [this](QUuid,const QList<FeatureTableEditResult*>& featureTableEditResults)
+  {
+    // A convenience wrapper that deletes the contents of featureEditResults when we leave scope.
+    FeatureTableEditResultsScopedCleanup featureTableEditResultsCleanup(featureTableEditResults);
+
+    for (auto featureTableEditResult : featureTableEditResults)
+    {
+      for (auto featureEditResult : featureTableEditResult->editResults())
+      {
+        if (!featureEditResult->error().isEmpty())
```

Comment:
I think it would be a nice improvement.

Given this example, I think it's a little too long and verbose where, `featureEditResult->error()`seems more direct and concise.